packages/headless/src/controllers/product-recommendations/headless-frequently-viewed-different-category.ts

```diff
@@ -0,0 +1,44 @@
+import {Schema, SchemaValues} from '@coveo/bueno';
+import {
+  baseProductRecommendationsOptionsSchema,
+  buildBaseProductRecommendationsList,
+} from './headless-base-product-recommendations';
+import {validateOptions} from '../../utils/validate-payload';
+import {ProductRecommendationEngine} from '../../app/product-recommendation-engine/product-recommendation-engine';
+
+const optionsSchema = new Schema({
+  ...baseProductRecommendationsOptionsSchema,
```

Comment:
Should this controller add any options or adjust certain ones to be required? Maybe `category` should be a required option?

Reply:
The objective was just to use a product's `permanentid` and show popular product from the same category. Exactly like `frequently-viewed-together` but scope to one category.